analyses/molecular-subtyping-MB/02-compare-classes.Rmd

```diff
@@ -0,0 +1,221 @@
+---
+title: "Comparison of Expected and Observed MB Subtype Classification"
+output:
+  html_document:
+    df_print: paged
+params:
+  expected_input:
+    value: input/expected_class.rds
+  observed_input:
+    value: results/mb-classified.rds
+---
+  
+```{r include = FALSE}
+knitr::opts_chunk$set(comment = NA)
+getOption(x = 'DT.warn.size', default = FALSE)
+```
+
+```{r load_packages, echo = TRUE}
+# load libraries
+suppressPackageStartupMessages(library(tidyverse))
+suppressPackageStartupMessages(library(knitr))
+suppressPackageStartupMessages(library(DT))
+```
+
+```{r read_data, echo = TRUE}
+# read input
+exp.class <- readRDS('input/expected_class.rds')
+obs.class <- readRDS('results/mb-classified.rds')
+
+# format ambiguous expected classes 
+exp.class <- exp.class %>%
+  mutate(pathology_subtype = replace(pathology_subtype, 
+                                     pathology_subtype == "Group 3 or 4", "Group3, Group4")) %>%
+  mutate(pathology_subtype = gsub(" ", "", pathology_subtype))
+```
+
+```{r dt_function, echo = FALSE}
+# custom datatable function
+viewDataTable <- function(dat){
+  DT::datatable(dat,
+                rownames = FALSE,
+                filter = "bottom",
+                class = 'cell-border stripe',
+                options = list(pageLength = 5,
+                               searchHighlight = TRUE,
+                               scrollX = TRUE,
+                               dom = 'tpi',
+                               initComplete = JS("function(settings, json) {",
+                                                 "$(this.api().table().header()).css({'background-color':
+                                            '#004467', 'color': '#fff'});","}"))
+  )
+}
+```
+
+#### Details:
+
+1. % Accuracy is currently being calculated by matching observed and expected subtypes where expected subtype info is available. In case of ambiguous subtypes, we treat it as a match if the observed subtype matches with any one of the expected subtypes.
+
+2. Consensus tabs: Molecular subtype labels are assigned only when the two classifiers agree, leaving all other samples as unclassified. 
+
+## {.tabset .tabset-fade}
+
+### MM2S (batch-corrected)
+
+```{r, echo = TRUE, warning = FALSE, message = FALSE}
+# merge observed and expected subtypes
+mm2s.corrected <- obs.class[[1]]
```

Comment:
Using an index here means we're relying on an analyst to know or remember what order the observed results come back in. There are no names in `obs.class` at the moment. I'd recommend making alterations to the upstream step (`01-classify-mb.R`) such that there is information about the method and dataset used in this object. That way someone who did not author the module could use this object "off-the-shelf" without much digging.

Reply:
Yes totally makes sense - will add names to the list items.

Reply:
added names to the list items: 
https://github.com/komalsrathi/OpenPBTA-analysis/blob/mb-class-nb/analyses/molecular-subtyping-MB/01-classify-mb.R#L49